### DISCUSSION TITLE
Problems with gov action details

### DISCUSSION THREAD

**Discussion by MSzalowski** [IntersectMBO/govtool#118] [General]
@jankun4 cite
We have a problem with gov action details. Previously we assumed that all details data will be in format like
```
{ 
 "some_key": 10,
 "some_other_key": ["value1", "value2"]
 "third_key": true
}
```
So no nested objects, and things like that. This assumption allowed us to render the details. Now however the details found in description field in db-sync are kind of chaotic. Here are some examples:
```
[null,{"anchor":{"dataHash":"12fbe9d5b89dc65026a922a402ad6d94a6eb2a060ae82df4b99f63513b4813e6","url":"http://bit.ly/ / PJt6U"}}]

null

[null,{"major":1,"minor":3}]

[null,[],{},0.5]

[[{"credential":{"keyHash":"830344068d6aaf09301dea1e338ff7eb48d1e478bd86698141219b9f"},"network":"Testnet"},40000000],[{"credential":{"keyHash":"ae157ac59fe9385dc9888330ecfe477d3d1f98c617866c
d3f61c073e"},"network":"Testnet"},10000000]]


[null,[],{"keyHash-a7e91bc521182701aa110f2eb5d29f5e3484a649cd3aa8eccdba156c":6247},0.2]

[{"govActionIx":0,"txId":"0ce78e5d0b2ff4079aa28a5d500e49b1f78ad35807656fae6a76aa11b4e3fcb4"},{"anchor":{"dataHash":"a84ac8c7407f513b65880afc847655bd02efc6e0aa0964f0eefb906b0822bd80","url":"ht
tp://localhost:3000/"}}]
```
Usually, the description for 2 gov actions with the same type is similar in shape, but I'm not sure if that's guarantee. Also can we be sure, that this data will be in this shape forever? 

@kickloop @jankun4

**Comment by Thomas-Upfield** [comment]
@kickloop to talk to the node team about this and resolve the answer.

**Reply by Ryun1** [reply]
would be DB-Sync team

**Comment by MSzalowski** [comment]
@jankun4 could you provide some example keys and their values?
eg:
```
{
   "govActionIx": 0,
   "description": "Test description"
}
```

Instead of something generic as it is here:
```
{ 
 "some_key": 10,
 "some_other_key": ["value1", "value2"]
 "third_key": true
}
```
